batch/batch/utils.py

```diff
@@ -193,10 +193,9 @@ def regions_to_bits_rep(selected_regions, all_regions_mapping):
 def regions_bits_rep_to_regions(regions_bits_rep, all_regions_mapping):
     if regions_bits_rep is None:
         return None
-    reverse_all_regions_mapping = {idx: region for region, idx in all_regions_mapping.items()}
     result = []
-    for i in range(64):
-        selected_region = bool((regions_bits_rep >> i) & 1)
+    for region, idx in all_regions_mapping.items():
+        selected_region = bool((regions_bits_rep >> idx - 1) & 1)
```

Comment:
I feel like this makes more sense to me as `regions_bits_rep & (1 << (idx - 1))` which I think is equivalent?

Reply:
More closely matches the function that goes in the other direction.

Reply:
It's not quite the same thing. You have to check whether your result is greater than 0. For example, if regions_bits_rep is 3 and idx is 2, then you get selected_region = 2. I'm not great at bit shift operations so I trust if this way makes more sense to you, then we should do that.